scipy/sparse/linalg/_traceest.py

```diff
@@ -0,0 +1,53 @@
+"""Trace estimator for LinearOperators."""
+import numpy as np
+
+from scipy.linalg.decomp_qr import qr
+
+
+__all__ = ["traceest"]
+
+
+def traceest(A, m3, seed=None):
```

Comment:
If this is a private function that's only needed in `exp_multiply`, then it would probably be better to add it in that file.

Reply:
I don't know if this might be in an interesting function to have for linear operators in general or not.

But as we don't have adequate tests so far, we probably shouldn't expose it at this point.